crates/rust-analyzer/src/config.rs

```diff
@@ -274,6 +274,8 @@ config_data! {
         inlayHints_renderColons: bool                              = "true",
         /// Whether to show inlay type hints for variables.
         inlayHints_typeHints_enable: bool                          = "true",
+        /// Whether to hide inlay type hints for initializations of closures with blocks.
+        inlayHints_typeHints_hideClosureInitialization: bool       = "false",
```

Comment:
```suggestion
        inlayHints_typeHints_hideClosureConstructor: bool       = "false",
```

Reply:
Yeah, the current wording does look a bit stilted to me. I'm not sure how to go about "closure constructor" though…

The thing is, strictly speaking Rust doesn't have "constructors" as a language concept, it's just customary to call them as such. If we consider Rust constructors as "associated functions that produce a value of the some type" (which is what it means for the adjacent `hideNamedConstructor` option), then calling the  [closure expression](https://doc.rust-lang.org/reference/expressions/closure-expr.html) a "closure constructor" seems a bit weird and unintuitive to me?

I mean, it somewhat makes sense if I think of it as "any expressions that produces a value", but literals like `42` and `"abc"` can be also considered a "`i32` constructor" and "`&'static str` constructor" then, which is not very helpful.

Reply:
You could argue that literals are constructors yes. I am very much opposed to using `initialization` here though as initializing a closure is not meaningful if you ask me.
And just because something isn't a language concept doesn't mean we can't use the term for it, constructors are very much an established concept in the language.

But sure, constructor may not be too fitting here. `hideClosureExpression` might be more fitting, but given HKalbasi's comment below I believe we might go with something else entirely here, I am not sure. (Sorry to be nagging about the config key here now, but given the recent config revamp I realized that we should put a lot more thought into configs :) )

Reply:
I agree that config keys can be important, and that `initialization` is not a very good name. I came up with it from a compiler internals perspective, where the "LetStatement" consists of an "irrefutable pattern" and an optional "initializer expression", so we are hiding the type hints when a variable is *initialized* to a closure value. But most end users probably won't know that/think of it like that, so calling it `hideClosureInitialization` doesn't convery the meaning very well.

And I think the same can be said for `hideClosureExpression` to a lesser extent — the phrase "closure expressions" doesn't come up anywhere outside the Rust reference. It may as well be `hideClosure` to the end users, which can also be somewhat confusing (hide closures *what?*).

Hmm, how about `hideClosureVariable` then?
```rust
/// Whether to hide inlay type hints for variables of a closure type.
/// Only applies to closures with blocks, same as `#rust-analyzer.inlayHints.closureReturnTypeHints.enable#`.
inlayHints_typeHints_hideClosureVariable: bool       = "false",
```

Reply:
Hmm, okay you know what. I think I like your first idea the most after all. Variable sounds too generic and I feel like we don't get something better than `hideClosureInitialization`. (And truth be told, if we don't get the name perfect here we can always change it without breaking things by simply supporting the old one).

The description should probably state that this specifically hides them for let statements though.

Reply:
One other thing I wonder though is if the polarity is right here. That is, should we have these be `hideX` with `false` as the default. `X_enable` with `true` as the default. I feel like the latter seems better almost everything else is also "positive".

Reply:
I get what you mean, though after some thinking I feel the current polarity happen to be the right one. Without `hideX` enabled, we are showing the type hints; with `hideX` enabled, we are showing the type hints *except for some special cases*. In my view, it's an *addition* to an exisiting feature instead of subtracting from it, despite "hide" being a negative word. So when I think about the feature known as "type inlay hints", I consider "everything enabled" to be it's "most complete form", if that makes sense.

Reply:
Ye I think you are right. I believe if we could split all type hints into proper categories that could be toggled on their own it would make sense to have them be "positive", but given we can't really do that here making these things "negative" (that is `hideX`) seems to make more sense.